engine.go

```diff
@@ -60,9 +66,39 @@ func (eng *Engine) Handlers() []Handler {
 // To prevent any deadlock from happening this method should never be called
 // from the handler's HandleMetric method.
 func (eng *Engine) Register(handler Handler) {
-	eng.hmutex.Lock()
+	eng.mutex.Lock()
 	eng.handlers = append(eng.handlers, handler)
-	eng.hmutex.Unlock()
+	eng.mutex.Unlock()
+}
+
+// HistogramBuckets returns a map of metric names to buckets used to distribute
+// hitogram values.
+//
+// The buckets are of list of upper limits used to group the observed values.
+func (eng *Engine) HistogramBuckets() map[string][]float64 {
+	eng.mutex.RLock()
+	buckets := make(map[string][]float64, len(eng.buckets))
+
+	for k, v := range eng.buckets {
+		buckets[k] = copyBuckets(v)
+	}
+
+	eng.mutex.RUnlock()
+	return buckets
+}
+
+// SetHistogramBuckets sets the buckets used for a histogram metric.
+//
+// Not all stats handler will respect the value distribution set with this
+// method, refer to the documentation of the handler for more details.
+func (eng *Engine) SetHistogramBuckets(name string, buckets ...float64) {
+	if !sort.Float64sAreSorted(buckets) {
+		panic("histogram buckets must be a sorted set of values")
```

Comment:
I don't think this is an error that the program should be recovering from, the list of buckets has to be sorted otherwise it's not a list of buckets.